frontend/src/Pages/StudentProfile.tsx

```diff
@@ -0,0 +1,287 @@
+                                        {
+                                            metrics?.session_engagement
+                                                .user_info.username
+                                        }
+                                    </div>
+                                    <div className="text-sm mt-2">
+                                        <span className="font-semibold">
+                                            Joined :
+                                        </span>{' '}
+                                        {new Date(
+                                            metrics.activity_engagement.first_active_date
+                                        ).toLocaleDateString('en-US')}
+                                    </div>
+                                    <div className="text-sm mt-2">
+                                        <span className="font-semibold">
+                                            Last Active :
+                                        </span>{' '}
+                                        {new Date(
+                                            metrics.activity_engagement.last_active_date
+                                        ).toLocaleDateString('en-US')}
+                                    </div>
+                                </div>
+                            </div>
+                        </div>
+                        {/* Chart */}
+                        <div className="flex-1 h-[240px] flex flex-col gap-4">
+                            <div className="card card-row-padding overflow-hidden">
+                                <h1 className="">
+                                    {' '}
+                                    {/* {metrics?.session_engagement.user_info.name_first +
+                                        " 's recent Activity"} */}
+                                </h1>
+                                <div className=" items-stretch">
+                                    <div className="h-[240px] overflow-visible">
+                                        <ResponsiveContainer
+                                            className="h-full p-7"
+                                            width="100%"
+                                            height="100%"
+                                            debounce={500}
+                                        >
+                                            <EngagementRateGraph
+                                                data={
+                                                    metrics?.session_engagement
+                                                        ?.user_engagement_times ??
+                                                    []
+                                                }
+                                                viewType="daily"
+                                            />
+                                        </ResponsiveContainer>
```

Comment:
I'm not far into the review for this yet, but can we make some changes to this graph:
- Fill out more of the container. There's a lot of padding around it.
- Depending on the width of the container, show more or less labels (See Operational Insights where the x-axis changes its values)
- If the days of activity is less than 30 days (which I assume is the range) then show only activity up to the number of days we have. On a fresh install I should only see activity for Day 1 and I shouldn't see labels for the other days since they haven't happened yet. 
- Add a header to the chart to indicate what we're showing here.
- Can the labels for the data points include the actual date and not Day 1, 2, 3, etc.?